tests/test_cmake_build/subdirectory_target/CMakeLists.txt

```diff
@@ -13,3 +13,11 @@ set_target_properties(test_cmake_build PROPERTIES PREFIX "${PYTHON_MODULE_PREFIX
 
 add_custom_target(check ${CMAKE_COMMAND} -E env PYTHONPATH=$<TARGET_FILE_DIR:test_cmake_build>
                   ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/../test.py ${PROJECT_NAME})
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND PYTHON_VERSION VERSION_LESS 3.0)
+  if(CMAKE_CXX_STANDARD VERSION_LESS 17)
+    target_compile_options(test_cmake_build PUBLIC -Wno-deprecated-register)
```

Comment:
Again, you can just use `LESS` above. Also, maybe skip the check and add both flags? Not sure about the latter.

Actually, if we know both of these are going to be numbers, can we use normal math symbols, like `>=`?

Reply:
No, math symbols are not used in CMake.

Reply:
But the following does print `less`.

```cmake
if(2 > 3)
  message("greater")
else()
  message("less")
endif()
```

Reply:
It's not listed here: https://cmake.org/cmake/help/latest/command/if.html#command:if

Reply:
And I don't see it listed here either: https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/cmConditionEvaluator.cxx

Reply:
And when I run the above code, I get:

```bash
$ cmake -P example.cmake
CMake Error at example.cmake:1 (if):
  if given arguments:

    "2" ">" "3"

  Unknown arguments specified
```

If I replace `>` with `LESS`, I get:


```bash
$ cmake -P example.cmake
greater
```

Reply:
(PS: I should have replaced it with GREATER, but not the point. :) )

Reply:
> It's not listed here: https://cmake.org/cmake/help/latest/command/if.html#command:if

I read that section about 5 times, thinking I might have missed something.

> cmake -P example.cmake

I was saving the example as `CMakeLists.txt` and running it as `cmake .`. 

> If I replace `>` with `LESS`, I get:

Good enough for me. I've got nothing more to complain about. Thanks for the pull request!

Reply:
Your complaints are always welcome. :)